### PR TITLE
Add competitve selection test with Zipf premium distribution

### PR DESCRIPTION
**Pull request by Kubuxu**

### CODE REVIEW

**Comment by whyrusleeping** [comment]
what does this do?

File: chain/messagepool/selection_test.go [hunk 1]
```diff
@@ -936,7 +936,7 @@ func testCompetitiveMessageSelection(t *testing.T, rng *rand.Rand) (float64, flo
 
 	capacityBoost := 0.0
 	rewardBoost := 0.0
-	const runs = 1
+	const runs = 50
```
Comment (vyzo) [inline chain/messagepool/selection_test.go:939]: we probably want to reset this for merge so that the test doesn't take 5min to run?

Comment (Kubuxu) [reply chain/messagepool/selection_test.go:939]: yeah

**Comment by Kubuxu** [comment]
@whyrusleeping it is just different approximation of message GasPremium distribution

**Comment by vyzo** [comment]
Made premium distribution parametric and we now have two tests for competitive selection: one for exponential distribution and one for Zipf; results don't change materially between the distributions with 50 runs, which is very confidence inspiring.

**Pull request merged**